### PR TITLE
Various minor fixes:

### PR DESCRIPTION
**Pull request by rmetzger**
- better exceptions for faulty temporary directories
- better exception if client can not connect to JobManager

### CODE REVIEW

**Comment by rmetzger** [comment]
Also, the PR correctly enforces maven 3.0.3.
I was unable to create a maven enforcer rule that fails on oraclejdk6. (there is a way by writing a custom rule in Java).

File: stratosphere-compiler/src/main/java/eu/stratosphere/compiler/PactCompiler.java [hunk 1]
```diff
@@ -1560,7 +1560,8 @@ private InstanceTypeDescription getInstanceTypeInfo() {
 			}
 		}
 		catch (Throwable t) {
-			throw new CompilerException("Available instances could not be determined from job manager: " + 
+			throw new CompilerException("Cannot connect to the JobManager to determine the available machines. "
```
Comment (fhueske) [inline stratosphere-compiler/src/main/java/eu/stratosphere/compiler/PactCompiler.java]: "available machines" -> "available TaskManagers"

**Comment by fhueske** [comment]
I have only one minor comment.
Otherwise good to merge.

**Comment by StephanEwen** [comment]
Agree, good to merge

**Pull request merged**